### DISCUSSION TITLE
Fast forward the display of events

### DISCUSSION THREAD

**Discussion by snapey** [spatie/ray#185] [Ideas]
If accidentally sending too much data to Ray (for instance when looping over models) the paced display of Ray lines can mean it can take a long time for the events to stop scrolling through the display.

It would be good to be able to 'fast forward' (removes all display delays) to the most recent event.

Current option is to close the window and start a fresh which is not good.

**Comment by freekmurze** [comment]
@AdrianMrn do you have an idea on how to solve this in a clean way?

**Reply by AdrianMrn** [reply]
I'm guessing this is related to https://github.com/spatie/ray/issues/184

I'll create an issue on the internal repository to track some possible performance improvements we could do.

**Comment by squatto** [comment]
I have a seeder that has a lot of log output when it runs, and I'm constantly running `migrate:fresh --seed` so it was blowing Ray up every single time. I worked around it by doing this before the process with the log output, to temporarily disable sending log calls to Ray:

```php
if (ray()->enabled()) {
    ray('Not sending log calls to Ray while seeding Applications...');
    ray()->settings->send_log_calls_to_ray = false;
}
```

And then this after the process was done:

```php
if (ray()->enabled()) {
    ray()->settings->send_log_calls_to_ray = true;
}
```

This obviously doesn't solve the issue in question, but it may help someone. You _could_ just disable Ray completely with `ray()->disable()`, but I didn't want to do that because I was still needing to send data to Ray, but didn't need to see all of the log calls.

At the very least, it may be helpful for people to know that you can change Ray's settings by directly accessing the `ray()->settings` property (which is actually just an instance of the `\Spatie\Ray\Settings\Settings` class that has `__set()` and `__get()` magic methods)
